deliver/lib/deliver/runner.rb

```diff
@@ -162,28 +162,25 @@ def upload_binary
       upload_ipa = options[:ipa]
       upload_pkg = options[:pkg]
 
-      # 2020-01-27
-      # Only verify platform if if both ipa and pkg exists (for backwards support)
-      if upload_ipa && upload_pkg
-        upload_ipa = ["ios", "appletvos"].include?(options[:platform])
-        upload_pkg = options[:platform] == "osx"
-      end
-
-      if upload_ipa
+      case options[:platform]
+      when "ios", "appletvos"
         package_path = FastlaneCore::IpaUploadPackageBuilder.new.generate(
           app_id: Deliver.cache[:app].id,
           ipa_path: options[:ipa],
           package_path: "/tmp",
           platform: options[:platform]
         )
-      elsif upload_pkg
+      when "osx"
         package_path = FastlaneCore::PkgUploadPackageBuilder.new.generate(
           app_id: Deliver.cache[:app].id,
           pkg_path: options[:pkg],
           package_path: "/tmp",
           platform: options[:platform]
         )
+      else
+        UI.user_error!("no suitable file found for upload for platform: #{options[:platform]}")
       end
+        
 
       transporter = transporter_for_selected_team
       result = transporter.upload(package_path: package_path, asset_path: upload_ipa || upload_pkg)
```

Comment:
Hey @snatchev  in this line the asset_path might still be wrong. For clarity i would duplicate this line in both 'switch' cases

Reply:
Something like this
```
 platform = options[:platform]
      transporter = transporter_for_selected_team
      
      case platform
        when "ios", "appletvos"
          package_path = FastlaneCore::IpaUploadPackageBuilder.new.generate(
            app_id: Deliver.cache[:app].id,
            ipa_path: options[:ipa],
            package_path: "/tmp",
            platform: platform
          )
          result = transporter.upload(package_path: package_path, asset_path: upload_ipa)

        when "osx"
          package_path = FastlaneCore::PkgUploadPackageBuilder.new.generate(
            app_id: Deliver.cache[:app].id,
            pkg_path: options[:pkg],
            package_path: "/tmp",
            platform: platform
          )
          result = transporter.upload(package_path: package_path, asset_path: upload_pkg)

        else
          UI.user_error!("no suitable file found for upload for platform: #{options[:platform]}")
        end

```

Reply:
@krzysiek84 this looks good to me, feel free to cook up a real PR for this change, and ping me for a review.